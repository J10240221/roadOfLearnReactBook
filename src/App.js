import React, { Component } from 'react';
import './App.css';

const list = [
  {
    title: 'React',
    url: 'www.baidu.com',
    author: 'Azhilingege',
    num_comments: 3,
    points: 4,
    objectID: 0,
  },
  {
    title: 'Redux',
    url: 'www.hao123.com',
    author: 'Bzhilingege',
    num_comments: 2,
    points: 5,
    objectID: 1,
  },
];
class App extends Component {
  constructor(props) {
    super(props);
    this.state = {
      list,
      searchTerm: '',
    };
    this.onSearchChange = this.onSearchChange.bind(this);
  }

  onDismiss = id => {
    const isNotId = item => item.objectID !== id;
    // filter、map、reduce 等都是纯函数式的，没有副作用，返回的也都是新的数组
    const updateList = this.state.list.filter(isNotId);
    this.setState({ list: updateList });
  };

  onSearchChange(e) {
    this.setState({ searchTerm: e.target.value });
  }

  render() {
    const { list: listState, searchTerm } = this.state; //使用结构赋值的方式
    const msg = 'welcome to the Road to learn React111';
    debugger;
    return (
      <div className="App">
        <h2>{msg}</h2>
        <form action="">
          输入查询名称：
          <input
            type="text"
            onChange={this.onSearchChange}
            value={searchTerm} //受控组件的写法
          />
        </form>
        {listState
          .filter(listItem =>
            listItem.title.toLowerCase().includes(searchTerm.toLowerCase())
          )
          .map(item => (
            <div key={item.objectID}>
              <span>
                <a href={item.url}>{item.title}</a>
              </span>
              <span>{item.author}</span>
              <span>{item.num_comments}</span>
              <span>{item.points}</span>
              <button onClick={() => this.onDismiss(item.objectID)}>
                Dismiss
              </button>
            </div>
          ))}
      </div>
    );
  }
}

export default App;
