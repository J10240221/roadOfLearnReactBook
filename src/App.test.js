import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';
import App, { Search, Button, Table } from './App';

describe('App', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App />, div);
    // ReactDOM.unmountComponentAtNode(div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<App />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Search', () => {
  it('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Search>Search</Search>, div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<Search>Search</Search>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Button', () => {
  it('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Button>More</Button>, div);
  });
  test('Button', () => {
    const component = renderer.create(<Button>more</Button>);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});

describe('Table', () => {
  const props = {
    list: [
      {
        title: 'React',
        url: 'www.baidu.com',
        author: 'Azhilingege',
        num_comments: 3,
        points: 4,
        objectID: 0,
      },
      {
        title: 'Redux',
        url: 'www.hao123.com',
        author: 'Bzhilingege',
        num_comments: 2,
        points: 5,
        objectID: 1,
      },
    ],
  };
  it('render without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<Table {...props} />, div);
  });

  test('has a valid snapshot', () => {
    const component = renderer.create(<Table {...props} />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
