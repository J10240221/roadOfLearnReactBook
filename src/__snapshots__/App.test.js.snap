// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App has a valid snapshot 1`] = `
<div
  className="page"
>
  <div
    className="interactions"
  >
    <form
      onSubmit={[Function]}
    >
      <input
        onChange={[Function]}
        type="text"
        value="redux"
      />
      <button
        type="submit"
      >
        搜索
      </button>
    </form>
  </div>
  <div
    className="table"
  >
    <div
      className="table-row"
    >
      <span
        style={
          Object {
            "width": "40%",
          }
        }
      >
        title
      </span>
      <span
        style={
          Object {
            "width": "30%",
          }
        }
      >
        author
      </span>
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
        num_comments
      </span>
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
        points
      </span>
      <span
        style={
          Object {
            "width": "10%",
          }
        }
      >
        operation
      </span>
    </div>
  </div>
  <div
    className="interactions"
  >
    <div>
      Loading...
    </div>
  </div>
</div>
`;

exports[`Button has a valid snapshot 1`] = `
<button
  className="button-inline"
  onClick={[Function]}
  type="button"
>
  more
</button>
`;

exports[`Search has a valid snapshot 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    onChange={[Function]}
    type="text"
    value=""
  />
  <button
    type="submit"
  >
    Search
  </button>
</form>
`;

exports[`Table has a valid snapshot 1`] = `
<div
  className="table"
>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      title
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
      author
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      num_comments
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      points
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      operation
    </span>
  </div>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <a
        href="www.baidu.com"
      >
        React
      </a>
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
      Azhilingege
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      3
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      4
    </span>
    <button
      className="button-inline"
      onClick={[Function]}
      type="button"
    >
      Dismiss
    </button>
  </div>
  <div
    className="table-row"
  >
    <span
      style={
        Object {
          "width": "40%",
        }
      }
    >
      <a
        href="www.hao123.com"
      >
        Redux
      </a>
    </span>
    <span
      style={
        Object {
          "width": "30%",
        }
      }
    >
      Bzhilingege
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      2
    </span>
    <span
      style={
        Object {
          "width": "10%",
        }
      }
    >
      5
    </span>
    <button
      className="button-inline"
      onClick={[Function]}
      type="button"
    >
      Dismiss
    </button>
  </div>
</div>
`;
